examples/tts/magpietts/evaluation.py

```diff
@@ -0,0 +1,152 @@
+# limitations under the License.
+"""
+Evaluation wrapper for MagpieTTS generated audio.
+
+This module provides a clean interface to the evaluation functionality,
+wrapping the existing `examples.tts.magpietts.evaluate_generated_audio` module.
+"""
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import Dict, List, Tuple
+
+# Import the existing evaluation module
+import examples.tts.magpietts.evaluate_generated_audio as evaluate_generated_audio
+import numpy as np
+import scipy.stats as stats
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class EvaluationConfig:
+    """Configuration for audio quality evaluation.
+
+    Attributes:
+        sv_model: Speaker verification model type ("titanet" or "wavlm").
+        asr_model_name: ASR model for transcription (e.g., "nvidia/parakeet-tdt-1.1b").
+        language: Language code for transcription (e.g., "en").
+        with_utmosv2: Whether to compute UTMOSv2 (Mean Opinion Score) metrics.
+    """
+
+    sv_model: str = "titanet"
+    asr_model_name: str = "nvidia/parakeet-tdt-1.1b"
+    language: str = "en"
+    with_utmosv2: bool = True
+
+
+def evaluate_generated_audio_batch(
+    manifest_path: str,
+    audio_dir: str,
+    generated_audio_dir: str,
+    config: EvaluationConfig,
+) -> Tuple[Dict[str, float], List[dict]]:
+    """Evaluate a batch of generated audio files against ground truth.
+
+    This function computes:
+    - ASR-based metrics: Character Error Rate (CER), Word Error Rate (WER)
+    - Speaker similarity: Cosine similarity using speaker embeddings
+    - Audio quality: UTMOSv2 scores (if enabled)
+
+    Args:
+        manifest_path: Path to the evaluation manifest (NDJSON format).
+        audio_dir: Directory containing ground truth audio files.
+        generated_audio_dir: Directory containing generated audio files.
+        config: Evaluation configuration.
+
+    Returns:
+        Tuple of:
+            - avg_metrics: Dictionary of averaged metrics across all files.
+            - filewise_metrics: List of per-file metric dictionaries.
+    """
+    logger.info(f"Evaluating generated audio from {generated_audio_dir} " f"against manifest {manifest_path}")
+
+    avg_metrics, filewise_metrics = evaluate_generated_audio.evaluate(
+        manifest_path=manifest_path,
+        audio_dir=audio_dir,
+        generated_audio_dir=generated_audio_dir,
+        language=config.language,
+        sv_model_type=config.sv_model,
+        asr_model_name=config.asr_model_name,
+        with_utmosv2=config.with_utmosv2,
+    )
+
+    return avg_metrics, filewise_metrics
+
+
+def compute_mean_with_confidence_interval(
+    metrics_list: List[dict],
+    metric_keys: List[str],
+    confidence: float = 0.95,
+) -> Dict[str, str]:
+    """Compute mean and confidence interval for specified metrics.
+
+    Uses Student's t-distribution for confidence interval calculation,
+    which is appropriate for small sample sizes.
```

Comment:
I would remove a subjective statement like "which is appropriate for small sample sizes."

T-test is appropriate when looking at independent, continuous, metrics that roughly follow a normal distribution. It cannot be used at all for non-continuous metrics like WER, CER.